presto-main/src/main/java/com/facebook/presto/sql/planner/optimizations/MergeWindows.java

```diff
@@ -85,30 +85,45 @@ protected PlanNode visitPlan(
                 RewriteContext<Multimap<WindowNode.Specification, WindowNode>> context)
```

Comment:
I didn't notice this comment before I approved this pull request, due to some github issues. I would fix this, but if you have strong opinion about it to not to change this code, then I am not going to block you. In case it got merged in this shape I will try to post PR to show you a fix and see your opinion.